### PR TITLE
Teacher

### PR DESCRIPTION
**Pull request by Ujsyinbin**
将teacher类的成员函数写完

### CODE REVIEW

**Comment by Ujsyinbin** [comment]
将其余地方的teacher函数写完

File: ExaminationManagementSystem/src/Database.cpp [hunk 1]
```diff
@@ -78,6 +78,8 @@ namespace EMS
     }
     Teacher* Database::get_teacher(IDType id) const
     {
+        Teacher newteacher(IDType id, const StrType& name, bool professor);
```
Comment (wxr001) [inline ExaminationManagementSystem/src/Database.cpp]: Remove these codes here.
This part has another person to care about.

**Comment by wxr001** [comment]
Where are the other get_teachers_* functions' implementation(ExamManagementSystem)

like get_teachers_by_date
get_teachers_by_duration
get_teachers_all

File: ExaminationManagementSystem/include/ExamManagementSystem.hpp [hunk 2]
```diff
@@ -41,7 +41,10 @@ namespace EMS
         }
         Classroom*   get_classroom(IDType classroom) const;
         Course*      get_course(IDType id) const;
-        Teacher*     get_teacher(IDType teacher) const;
+        Teacher*     get_teacher(IDType teacher) const{
```
Comment (wxr001) [inline ExaminationManagementSystem/include/ExamManagementSystem.hpp]: Do not add codes in headers.
Move these implementation to cpp files.
And use Database::get_instance()->get_teacher(teacher) to implement this function.

File: ExaminationManagementSystem/src/Database.cpp (hunk 1 above)
Comment (wxr001) [reply ExaminationManagementSystem/src/Database.cpp]: DO NOT CHANGE ANY OTHER CODES THAT IS NOT YOUR PART!!!

**Comment by wxr001** [comment]
Do not change the codes outside of the TeacherImpl class.
Just finish the codes in TeacherImpl class.
Move all your changes in Teacher.cpp to TeacherImpl class.
And remove those changes.

**Comment by Ujsyinbin** [comment]
finished your request

File: ExaminationManagementSystem/src/ExamManagementSystem.cpp [hunk 3]
```diff
@@ -19,4 +19,17 @@
 
 namespace EMS
 {
+    Teacher* Teacher::get_teacher(IDType teacher) const{
+        Database::get_instance()->get_teacher(teacher);
```
Comment (wxr001) [inline ExaminationManagementSystem/src/ExamManagementSystem.cpp]: Where is `return`

File: ExaminationManagementSystem/src/Teacher.cpp [hunk 4]
```diff
@@ -23,16 +23,138 @@ namespace EMS
...
+            }
+            p=course_head;
+            q=new CourseImpl(IDType id,const StrType& name,IDType teacher, unsigned int capacity);//构造节点
+            if(course_length==0){//判断链表是否为空
+                p->next=q;
+            }
+            else for(;p->next!=NULL;p=p->next);//不为空则在表尾添加节点
+            p->next=q;
+            course_length++;
+            cout<<"Add success";
```
Comment (wxr001) [inline ExaminationManagementSystem/src/Teacher.cpp]: > cout

 here is no meaning.
Remove it.

**Comment by wxr001** [comment]
1、不要使用cout
2、你可以假定在Teacher.cpp里所有函数的输入都是有效的，即不用在add_course里检查某门课程是否已经被添加过。remove也同理。
3、不要使用CourseImpl或是其他的Impl类，而是应该存储对应的ID值，然后在需要的时候（如在get_course_list里）通过ID值和Database类来间接的获取对应的指针。
4、ListType<T>的使用方法在项目说明里有，你在代码里的用法是错的，请重写get_course_list()/get_exam_list()
5、Exam部分也是只存储ID，通过ID值和Database类来间接的获取对应的指针。
6、对于get_teachers_all()，Database里有get_teacher_list函数来实现类似的操作，可以通过调用get_teacher_list函数来实现相关的操作。

**Comment by Ujsyinbin** [comment]
finished

**Pull request merged**